File: solutions/ruby/problem-00001.rb
#!/usr/bin/env ruby
# [minhuang@mocha:ruby]$ time ruby problem-00001.rb 
# 233168
#
# real    0m0.016s
# user    0m0.012s
# sys     0m0.008s
result = (1..999).select { |x| x % 3 == 0 || x % 5 == 0 }.inject(:+)
puts result

File: src/javascript/problem-00001.js
// [minhuang@mocha:javascript(master)]$ time js -f problem-00001.js
// 233168
//
// real 0m0.009s
// user 0m0.008s
// sys  0m0.004s
function range(begin, end) {
  for (let i = begin; i < end; i++) yield i;
}

let xs = range(1, 1000);
let result = [x for each (x in xs) if (x % 3 === 0 || x % 5 === 0)].
  reduce(function (a, b) {
    return a + b;
  });
print(result);
